registry.go

```diff
@@ -35,6 +36,9 @@ type Namespace interface {
 	// registry may or may not have the repository but should always return a
 	// reference.
 	Repository(ctx context.Context, name string) (Repository, error)
+
+	// Catalog returns a reference which can be used for listing repositories
+	Catalog(ctx context.Context, driver storagedriver.StorageDriver) CatalogService
```

Comment:
The `namespace` has a driver (or some other storage engine). Driver should not be an argument here.